### ISSUE TITLE
Error trying to use conda env by name if Snakemake is not installed in base

### ISSUE THREAD

**Issue by tdido** [snakemake/snakemake#2288]
**Snakemake version**
\>=7.25.0

**Describe the bug**
Snakemake fails to find existing conda environments by name if Snakemake itself is installed in an environment other than `base`.

**Minimal example**
[conda_by_name_error.tar.gz](https://github.com/snakemake/snakemake/files/11674978/conda_by_name_error.tar.gz)

Steps to reproduce:
1. Create conda environments using included yaml files
2. Activate `snakemake` environment
3. Execute workflow

This will result in an error like this:

```
EnvironmentNameNotFound: Could not find conda environment: samtools
You can list all discoverable environments with `conda info --envs`.
```

**Additional context**
Although this does not happen with Snakemake <= 7.24.2, if you downgrade Snakemake after having >=7.25.0 installed the problem persists. Only workaround is a fresh environment with v7.24.2.

**Comment by dariober** [comment]
I also notice that if `samtools` (or whatever tool) is on the search PATH, snakemake will throw the `EnvironmentNameNotFound` error but it will still proceed and complete without errors.

**Comment by tdido** [comment]
Right, I think I've found the cause of this. `mamba` was added to the bioconda recipe dependencies in [build 1 of the v7.25.0 recipe](https://github.com/bioconda/bioconda-recipes/commit/c19e07478b7d91b691d35080ce73a475cbccbfdb). This also pulls the `conda` package into the environment.

When you have conda installed in a non-base environment, `conda info` will treat that environment as "base". When Snakemake gets the environment path [here](https://github.com/snakemake/snakemake/blob/c767a855ccafb762672d859a3f4c453e205f2c23/snakemake/deployment/conda.py#L660) the resulting path will be the one of the subenvironment where Snakemake is installed rather than that of `base`. `source activate` using the subenvironment's path fails and that is the source of the error.

I suppose this case of conda being present in the Snakemake environment is still something that should preferably be handled in the Snakemake code? Or is it just a matter of removing mamba from the recipe?
